src/error.rs

```diff
@@ -61,28 +76,28 @@ impl fmt::Display for Error {
 
 impl<E: error::Error + 'static> From<E> for Error {
     fn from(err: E) -> Self {
-        Error::new(&err.to_string()).context(any::type_name::<E>())
+        Error::new(&err.to_string()).set_context(any::type_name::<E>())
     }
 }
 
 pub trait ResultError {
-    fn source(self, source: &str) -> Self;
+    fn set_source(self, source: &str) -> Self;
 
-    fn cause(self, cause: impl error::Error + 'static) -> Self;
+    fn set_cause(self, cause: impl error::Error + 'static) -> Self;
 
-    fn context(self, context: &str) -> Self;
+    fn set_context(self, context: &str) -> Self;
 }
 
 impl<T> ResultError for Result<T> {
-    fn source(self, source: &str) -> Self {
-        self.map_err(|e| e.source(source))
+    fn set_source(self, source: &str) -> Self {
```

Comment:
Just asking what the rationale for changing it to `set_*` was as we discussed last time that this wasn't the convention?

Reply:
Convention is to use use getter `source` and setter `set_source` but not `get_source` for getters.

Reply:
https://doc.rust-lang.org/1.0.0/style/style/naming/README.html
> A method foo(&self) -> &T for getting the current value of the field.

> A method set_foo(&self, val: T) for setting the field. (The val argument here may take &T or some other type, depending on the context.)